xarray/plot/plot.py

```diff
@@ -750,6 +767,13 @@ def _infer_interval_breaks(coord, axis=0):
            [ 2.5,  3.5,  4.5]])
     """
     coord = np.asarray(coord)
+
+    if not _is_monotonic(coord, axis=axis):
+        warnings.warn("The input coordinate is not sorted in increasing order "
```

Comment:
Rather an a warning, why not make this an error? I don't see any use-cases for 2d plots with non-monotonic coordinates. With the current version of xarray, these plots always end up wrong in some way, either by not plotting everything or with bad axis labels.